### ISSUE TITLE
Payment getting processed multiple times

### ISSUE THREAD

**Issue by Divya-motivitylabs** [AuthorizeNet/sdk-dotnet#202]
Once Failed, payment is getting processed multiple times. Exception has been received from authorize.net

" Server in maintenance "

**Comment by gnongsie** [comment]
Can you provide the URL you are sending the requests to? Is this Sandbox or Production?

**Comment by Divya-motivitylabs** [comment]
Its the production URL ,and the customer profile id is 215391028

**Comment by gnongsie** [comment]
Hi Divya, I had waited to see if any other complaints came in through Customer Support about the server being in maintenance. But we did not receive any calls for this. Can you confirm your request and credentials?

**Comment by Divya-motivitylabs** [comment]
Yes, everything is correct, the transaction_id for the failure request is 1507716001829.
Is there any service from which i can see all the transactions for a user??

**Comment by Divya-motivitylabs** [comment]
Hi, 
I could see almost 24 transactions for the same day with status voided in authorize.net

5214_1507860001594   voided   2017-10-13T02:00:02.013Z
5214_1507856402200   voided   2017-10-13T01:00:02.64Z
5214_1507852801676   voided   2017-10-13T00:00:02.137Z
5214_1507849201620   voided   2017-10-12T23:00:02.097Z
5214_1507845602892   voided   2017-10-12T22:00:03.377Z
5214_1507842001823   voided   2017-10-12T21:00:02.28Z
5214_1507838401612   voided   2017-10-12T20:00:01.983Z
5214_1507834801695   voided   2017-10-12T19:00:02.117Z
5214_1507831201653   voided   2017-10-12T18:00:02.223Z
5214_1507827601825   voided   2017-10-12T17:00:02.233Z
5214_1507824001674   voided   2017-10-12T16:00:02.453Z
5214_1507820401677   voided   2017-10-12T15:00:02.18Z
5214_1507816801737   voided   2017-10-12T14:00:02.13Z
5214_1507813202592   voided   2017-10-12T13:00:03.497Z
5214_1507809601601   voided   2017-10-12T12:00:02.15Z
5214_1507806001695   voided   2017-10-12T11:00:02.86Z
5214_1507802401653   voided   2017-10-12T10:00:02.16Z
5214_1507798801569   voided   2017-10-12T09:00:02.457Z
5214_1507795202015   voided   2017-10-12T08:00:02.423Z
5214_1507791602032   voided   2017-10-12T07:00:02.43Z
5214_1507788001656   voided   2017-10-12T06:00:02.387Z
5214_1507784401749   voided   2017-10-12T05:00:02.04Z

But at our end, I have 5214_1507860001594 as invoice number. Also the service call is only once a day in our system. how is it possible that it happens every hour

**Comment by Divya-motivitylabs** [comment]
Also, the transaction for which i was getting "Server in maintenance", its invoice number is 5214_1507716001829 and error code is 2

**Comment by gnongsie** [comment]
You can get transactions for a merchant through the Merchant Interface. There are APIs that you can also use: GetTransactionList and GetUnsettledTransactionList.

This seems like your service call is going through a retry cycle. Can you verify that the service call is only made once? It may be that you trigger it once but the code is designed to perform a retry.

As it is, I am forwarding this issue to the Customer Support to see if they have dealt with this kind of behavior before. Since we are not getting any other complaints about this, I don't think this is an issue with the server code.

**Comment by gnongsie** [comment]
Can I also ask, if you are recording 5214_1507860001594 as the invoice number, how is it possible that there were requests sent before that? It seems like your retry logic is making a request every hour and recording the last invoice after it has failed. Can you confirm that you have properly implemented the calls?

**Comment by Divya-motivitylabs** [comment]
There is no retry logic at our end, we use to try processing it next day, not on the same day again.
Also can you give me the the details of 5214_1507716001829 invoice, as it is happening after this request got failed.

**Comment by Divya-motivitylabs** [comment]
Also, can you tell me the significance of status as 'voided'? if payment got success it is 'settledSuccessfully'

**Comment by gnongsie** [comment]
I don't think you can search for a transaction based on invoice number through our APIs. You can try to do that through Merchant Interface.

- A transaction that enters our system can be 'authorized and captured' or 'declined'. A declined transaction basically means something is wrong with the request.
- After the transaction is authorized and captured, it shows up as 'Unsettled', while it waits for the settlement process to complete. After settlement, the transaction goes to 'settledSuccessfully' state.
- You can void a transaction when it is in 'Unsettled' state. This basically means to cancel the transaction so that no money changes hands (during settlement).

Can you provide us with the code you are using to make these requests? Perhaps, send it in a zipped folder to gnongsie@visa.com .

**Comment by Purushottamlpa** [comment]
Thanks for the concern. Since, we didn't hear any update from you, we are closing this issue. If you need further assistance please raise a new issue.
